src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7912,13 +7836,24 @@ internal static bool IsSameFileSystemItem(string pathOne, string pathTwo)
 #if !UNIX
         private static bool WinIsSameFileSystemItem(string pathOne, string pathTwo)
         {
-            const FileAccess access = FileAccess.Read;
-            const FileShare share = FileShare.Read;
-            const FileMode creation = FileMode.Open;
-            const FileAttributes attributes = FileAttributes.BackupSemantics | FileAttributes.PosixSemantics;
-
-            using (var sfOne = AlternateDataStreamUtilities.NativeMethods.CreateFile(pathOne, access, share, IntPtr.Zero, creation, (int)attributes, IntPtr.Zero))
-            using (var sfTwo = AlternateDataStreamUtilities.NativeMethods.CreateFile(pathTwo, access, share, IntPtr.Zero, creation, (int)attributes, IntPtr.Zero))
+            const int FILE_FLAG_BACKUP_SEMANTICS = 0x02000000;
+            const int FILE_FLAG_POSIX_SEMANTICS = 0x01000000;
+            SafeFileHandle sfOne = File.OpenHandle(
+                pathOne,
+                FileMode.Open,
+                FileAccess.Read,
+                FileShare.Read,
+                (FileOptions)(FILE_FLAG_BACKUP_SEMANTICS | FILE_FLAG_POSIX_SEMANTICS));
```

Comment:
Is this actually something publicly supported or just an implementation detail that they pass through to the flags and attributes parameter on `CreateFileW`. Can we rely on the fact that this will always flow through or do we need to ask the dotnet team to add these options to the enum so it's no longer just an implementation detail that we are relying on.

Reply:
It is not supported in the API. 😕 I have to revert this.

Reply:
That's a pity, hopefully they introduce these options as official enums as I've always wanted to be able to easily specify the backup flag without resorting to PInvoke.

Reply:
@jborean93 I opened feature request https://github.com/dotnet/runtime/issues/79335 Please vote/comment there.